dataRetriever/factory/resolverscontainer/shardResolversContainerFactory.go

```diff
@@ -0,0 +1,303 @@
+package resolverscontainer
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core/random"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	triesFactory "github.com/ElrondNetwork/elrond-go/data/trie/factory"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/factory/containers"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/resolvers"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/resolvers/topicResolverSender"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+var _ dataRetriever.ResolversContainerFactory = (*shardResolversContainerFactory)(nil)
+
+type shardResolversContainerFactory struct {
+	*baseResolversContainerFactory
+}
+
+// NewShardResolversContainerFactory creates a new container filled with topic resolvers for shards
+func NewShardResolversContainerFactory(
+	shardCoordinator sharding.Coordinator,
```

Comment:
an arg would reduce the dupplicate parameters

Reply:
done